src/Lucene.Net/Index/SegmentMerger.cs

```diff
@@ -44,6 +44,38 @@ namespace Lucene.Net.Index
 	/// </seealso>
 	public sealed class SegmentMerger
 	{
+		private class OptimizeScopeCheckAbort : CheckAbort
+		{
+			private readonly OptimizeScope _scope;
+
+			public OptimizeScopeCheckAbort(MergePolicy.OneMerge merge, Directory dir, OptimizeScope scope) : base(merge, dir)
+			{
+				_scope = scope;
+			}
+
+			/// <summary> Records the fact that roughly units amount of work
+			/// have been done since this method was last called.
+			/// When adding time-consuming code into SegmentMerger,
+			/// you should test different values for units to ensure
+			/// that the time in between calls to merge.checkAborted
+			/// is up to ~ 1 second.
+			/// </summary>
+			public override void Work(double units, IState state)
+			{
+				workCount += units;
+				if (workCount >= 10000.0)
```

Comment:
It is the same logic as in the base class, right?

Reply:
It has an additional check related to CancellationToken.